doc/source/tune-config.rst

```diff
@@ -0,0 +1,78 @@
+Experiment Configuration
+========================
+
+
+Experiment Setup
+----------------
+
+There are two ways to setup an experiment - one via Python and one via JSON.
+
+The first is to create an Experiment object. You can then pass in either
+a single experiment or a list of experiments to `run_experiments`, as follows:
+
+.. code-block:: python
+
+    # Single experiment
+    run_experiments(Experiment(...))
+
+    # Multiple experiments
+    run_experiments([Experiment(...), Experiment(...), ...])
+
+.. autoclass:: ray.tune.Experiment
+
+An example of this can be found in `hyperband_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/hyperband_example.py>`__.
+
+Alternatively, you can pass in a JSON object. This uses the same fields as
+the `ray.tune.Experiment`, except the experiment name is the key of the top level
+dictionary.
+
+.. code-block:: python
+
+    run_experiments({
+        "my_experiment_name": {
+            "run": "my_func",
+            "resources": { "cpu": 1, "gpu": 0 },
+            "stop": { "mean_accuracy": 100 },
+            "config": {
+                "alpha": grid_search([0.2, 0.4, 0.6]),
+                "beta": grid_search([1, 2]),
+            },
+            "upload_dir": "s3://your_bucket/path",
+            "local_dir": "~/ray_results",
+            "max_failures": 2
+        }
+    })
+
+An example of this can be found in `async_hyperband_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/async_hyperband_example.py>`__.
+
+
+Trial Variant Generation
+------------------------
+
+In the above example, we specified a grid search over two parameters using the ``grid_search`` helper function. Ray Tune also supports sampling parameters from user-specified lambda functions, which can be used in combination with grid search.
+
+The following shows grid search over two nested parameters combined with random sampling from two lambda functions. Note that the value of ``beta`` depends on the value of ``alpha``, which is represented by referencing ``spec.config.alpha`` in the lambda function. This lets you specify conditional parameter distributions.
+
+.. code-block:: python
+
+    "config": {
+        "alpha": lambda spec: np.random.uniform(100),
+        "beta": lambda spec: spec.config.alpha * np.random.normal(),
```

Comment:
Not sure about this, but I was using some conditional distributions akin to this to tune # of fully connected layers in a deep network. Something like 
```
"config": {
    "fc1": lambda spec: np.random.choose([10, 100]),
    "fc2": lambda spec: spec.config.fc1 // 2,
...
}
```
and I was seeing trials with `alpha = 10` and `beta = 10`. Not sure, but it could be the case that a lambda is being evaluated multiple times. Could also be a bug on my end.

Reply:
oh weird - could you post an example as an issue?